.github/workflows/documentation-creation.yml

```diff
@@ -0,0 +1,51 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Doxygen Action
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches: [ doc-upload ]
+    #paths:
+    #  - 'xbmc/addons/kodi-dev-kit/**'
+ 
+jobs:
+  build:
+    if: github.repository == 'AlwinEsch/kodi'
```

Comment:
The `branches: [ doc-upload ]` above should be in Kodi `branches: [ master ]` (currently set to my one for test).
And `if: github.repository == 'AlwinEsch/kodi'` should be then `if: github.repository == 'xbmc/xbmc`

As working upload by script I has on begin to here https://github.com/AlwinEsch/kodi/tree/gh-pages where works on a forked branch.